narwhal/primary/src/proposer.rs

```diff
@@ -114,6 +119,21 @@ impl Proposer {
         .await;
         debug!("Created {header:?}");
 
+        // Equivocation protection using the proposer store
```

Comment:
Do we have to do this check every time we produce a header or can we just do it every time we reboot?

Reply:
We should do it every time because it enforces safety more generally. We are also going to need to store each header regardless since we don't know when the next reboot will be.